types/options.go

```diff
@@ -114,11 +120,50 @@ func loadDefaultStoreOptions() {
 
 // loadStoreOptions returns the default storage ops for containers
 func loadStoreOptions() (StoreOptions, error) {
-	storageConf, err := DefaultConfigFile()
+	baseConf, err := DefaultConfigFile()
+	if err != nil {
+		return defaultStoreOptions, err
+	}
+
+	// Load the base config file
+	baseOptions, err := loadStoreOptionsFromConfFile(baseConf)
+	if err != nil {
+		return defaultStoreOptions, err
+	}
+
+	if _, err := os.Stat(defaultDropInConfigDir); err != nil && os.IsNotExist(err) {
+		return defaultStoreOptions, err
+	}
+
+	err = mergeConfigFromDirectory(&baseOptions, defaultDropInConfigDir)
 	if err != nil {
 		return defaultStoreOptions, err
 	}
-	return loadStoreOptionsFromConfFile(storageConf)
+
+	return baseOptions, nil
+}
+
+func mergeConfigFromDirectory(baseOptions *StoreOptions, configDir string) error {
+	err := filepath.Walk(configDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
+		}
+
+		// Load drop-in options from the current file
+		err = ReloadConfigurationFile(path, baseOptions)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
```

Comment:
return filepath.Walk(...)... The if statements are not necessary.

Reply:
Thanks. updated.